app/services/mfa/management_service.rb

```diff
@@ -0,0 +1,89 @@
+class Mfa::ManagementService
+  pattr_initialize [:user!]
+
+  def enable_two_factor!
+    user.otp_secret = User.generate_otp_secret
+    user.save!
+  end
+
+  def disable_two_factor!
+    user.otp_secret = nil
+    user.otp_required_for_login = false
+    user.otp_backup_codes = nil
+    user.save!
+  end
+
+  def verify_and_activate!
+    ActiveRecord::Base.transaction do
+      user.update!(otp_required_for_login: true)
+      backup_codes_generated? ? nil : generate_backup_codes!
+    end
+  end
+
+  def provisioning_uri
+    return nil if user.otp_secret.blank?
+
+    issuer = 'Chatwoot'
+    label = user.email
+    user.otp_provisioning_uri(label, issuer: issuer)
+  end
+  alias two_factor_provisioning_uri provisioning_uri
+
+  def generate_backup_codes!
+    codes = Array.new(10) { format('%08d', SecureRandom.random_number(100_000_000)) }
```

Comment:
@sojan-official need your thoughts, shouldn't the backup codes be alphanumeric... 

```suggestion
    codes = Array.new(10) { SecureRandom.hex(4).upcase }
```

Reply:
any specific reasons ?  @tds-1  ? if not lets have them alphanumeirc